File: JavaScript/StudyPlan/LeetCode75/Q1768-MergeStringsAlternately.js
/**
 * @param {string} word1
 * @param {string} word2
 * @return {string}
 */
var mergeAlternately = function(word1, word2) {
    let wordOneLen = word1.length;
    let wordTwoLen = word2.length;

    let mergedStr = "";
    let i = 0, j = 0;
    while(i < wordOneLen && j < wordTwoLen) {
        mergedStr += word1[i];
        mergedStr += word2[j];
        i++, j++;
    }

    while(i < wordOneLen) {
        mergedStr += word1[i];
        i++;
    }

    while(j < wordTwoLen) {
        mergedStr += word2[j];
        j++;
    }

    return mergedStr;
};


const mergeAlternately = (word1, word2) => {
    let str = "";
    for(let i = 0; i < Math.max(word1.length, word2.length); i++) {
        if(i < word1.length) str += word1[i];
        if(i < word2.length) str += word2[i];
    }

    return str;
}

File: Python/Leetcode75/Easy/String/Q1768-MergeStringsAlternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        merged_str = ""
        n, m = len(word1), len(word2)

        if n > m:
            for i in range(m):
                merged_str += word1[i]
                merged_str += word2[i]
            
            merged_str += word1[i+1:]
        else:
            for i in range(n):
                merged_str += word1[i]
                merged_str += word2[i]

            merged_str += word2[i+1:]

        return merged_str
    
    def mergeAlternatelyOptimized(self, word1: str, word2: str) -> str:
        merged_str = ""

        for i in range(max(len(word1), len(word2))):
            if i < len(word1):
                merged_str += word1[i]
            if i < len(word2):
                merged_str += word2[i]

        return merged_str